File: src/main/intellibitz/sted/actions/TableModelListenerAction.java
package intellibitz.sted.actions;

import javax.swing.event.ListSelectionEvent;
import javax.swing.event.TableModelEvent;
import javax.swing.event.TableModelListener;
import javax.swing.table.TableModel;

abstract public class TableModelListenerAction
        extends TableRowsSelectAction
        implements TableModelListener {
    protected TableModelListenerAction() {
        super();
    }

    /**
     * This fine grain notification tells listeners the exact range of cells,
     * rows, or columns that changed.
     */
    public void tableChanged(TableModelEvent e) {
        setEnabled(((TableModel) e.getSource()).getRowCount() > 0);
    }

    public void valueChanged(ListSelectionEvent e) {
    }

}

File: src/main/intellibitz/sted/actions/TableModelListenerAction.kt
package sted.actions

import javax.swing.event.ListSelectionEvent
import javax.swing.event.TableModelEvent
import javax.swing.event.TableModelListener
import javax.swing.table.TableModel

abstract class TableModelListenerAction protected constructor() : TableRowsSelectAction(), TableModelListener {
    /**
     * This fine grain notification tells listeners the exact range of cells,
     * rows, or columns that changed.
     */
    override fun tableChanged(e: TableModelEvent) {
        isEnabled = (e.source as TableModel).rowCount > 0
    }

    override fun valueChanged(e: ListSelectionEvent) {}
}